components/datadog/apps/fakeintake/fargateFakeintakeService.go

```diff
@@ -54,164 +55,25 @@ func NewECSFargateInstance(e aws.Environment, option ...fakeintakeparams.Option)
-		}, opts...)
-		if err != nil {
-			return nil, err
-		}
-
-		current, err := paws.GetPartition(e.Ctx, nil, nil)
-		if err != nil {
-			return nil, err
-		}
-		caCert, err := acmpca.NewCertificate(e.Ctx, namer.ResourceName("ca-cert"), &acmpca.CertificateArgs{
-			CertificateAuthorityArn:   ca.Arn,
-			CertificateSigningRequest: ca.CertificateSigningRequest,
-			SigningAlgorithm:          pulumi.String("SHA512WITHRSA"),
-			TemplateArn:               pulumi.String(fmt.Sprintf("arn:%v:acm-pca:::template/RootCACertificate/V1", current.Partition)),
-			Validity: acmpca.CertificateValidityArgs{
-				Value: pulumi.String("10"),
-				Type:  pulumi.String("YEARS"),
-			},
-		}, opts...)
-		if err != nil {
-			return nil, err
-		}
-
-		if _, err = acmpca.NewCertificateAuthorityCertificate(e.Ctx, namer.ResourceName("ca-cert-cert"), &acmpca.CertificateAuthorityCertificateArgs{
-			CertificateAuthorityArn: ca.Arn,
-			Certificate:             caCert.Certificate,
-			CertificateChain:        caCert.CertificateChain,
-		}, opts...); err != nil {
-			return nil, err
-		}
-
-		domainName := alb.LoadBalancer.DnsName().ApplyT(func(dnsName string) *string {
-			if len(dnsName) <= 64 {
-				return &dnsName
-			}
-			dummyDNSName := "fakeintake.datadoghq.com"
-			return &dummyDNSName
-		}).(pulumi.StringPtrOutput)
-		cert, err := acm.NewCertificate(e.Ctx, namer.ResourceName("cert"), &acm.CertificateArgs{
-			CertificateAuthorityArn: ca.Arn,
-			KeyAlgorithm:            pulumi.String("RSA_2048"),
-			DomainName:              domainName,
-			SubjectAlternativeNames: pulumi.StringArray{
-				alb.LoadBalancer.DnsName(),
-			},
-		}, opts...)
-		if err != nil {
-			return nil, err
-		}
 
-		if _, err = clb.NewListener(e.Ctx, namer.ResourceName("lb-https"), &clb.ListenerArgs{
-			LoadBalancerArn: alb.LoadBalancer.Arn(),
-			Port:            pulumi.IntPtr(sslPort),
-			Protocol:        pulumi.StringPtr("HTTPS"),
-			CertificateArn:  cert.Arn,
-			DefaultActions: clb.ListenerDefaultActionArray{
-				clb.ListenerDefaultActionArgs{
-					Type:           pulumi.String("forward"),
-					TargetGroupArn: alb.DefaultTargetGroup.Arn(),
-				},
-			},
-		}, opts...); err != nil {
-			return nil, err
-		}
+	taskDef, err := fargateLinuxTaskDefinition(e, namer, params.ImageURL)
+	if err != nil {
+		return nil, err
+	}
 
-		instance.Host = alb.LoadBalancer.DnsName()
-		balancerArray = classicECS.ServiceLoadBalancerArray{
-			&classicECS.ServiceLoadBalancerArgs{
-				ContainerName:  pulumi.String(containerName),
-				ContainerPort:  pulumi.Int(port),
-				TargetGroupArn: alb.DefaultTargetGroup.Arn(),
-			},
-		}
+	if params.LoadBalancerEnabled {
+		instance.Host, err = fargateServiceFakeIntakeWithLoadBalancer(e, namer, taskDef, opts...)
 	} else {
-		instance.Host, err = fargateServiceFakeintakeWithoutLoadBalancer(e, params.Name, params.ImageURL)
+		instance.Host, err = fargateServiceFakeintakeWithoutLoadBalancer(e, namer, taskDef)
 		if err != nil {
 			return nil, err
 		}
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You can move this err check out of the if